api/app/Models/User.php

```diff
@@ -251,6 +254,28 @@ public static function scopeIsProfileComplete(Builder $query, ?bool $isProfileCo
     protected static function boot()
     {
         parent::boot();
+
+        static::deleting(function (User $user) {
+            // Cascade delete to child models
+            foreach ($user->poolCandidates as $candidate) {
+                $candidate->delete();
+            }
+
+            // Modify the email to allow it to be used for another user
+            $newEmail = $user->email . "-deleted-at-" . Carbon::now()->format('Y-m-d');
+            $user->update(['email' => $newEmail]);
+        });
+
+        static::restoring(function (User $user) {
+            // Cascade restore to child models
+            foreach ($user->poolCandidates()->withTrashed()->get() as $candidate) {
+                $candidate->restore();
+            }
+
+            // Remove the deleted-at text from the end of the email
+            $newEmail = substr($user->email, 0, -22);
+            $user->update(['email' => $newEmail]);
+        });
```

Comment:
What if you try and restore an account and the email is being used by another account, say
Account made by user with email X. 
Account deleted for whatever reason, accidental or otherwise. Set to X-deleted. 
User tries to get account restored, in the meanwhile makes an account with freed email X. 
Admin cannot restore the account anymore as the -deleted can't be removed without violating uniqueness constraint

Reply:
Agreed, the user can restore their own email after the account is restored.

Reply:
I've updated this to specifically error out if the users email would be duplicated. This isn't actually used on the frontend currently though.

Reply:
If it would collide, maybe don't adjust the email instead of just cancelling the operation?

Reply:
I prefer peter's first instinct - let's not remove anything automatically. I'm worried about that being a bit brittle. If an admin restores a user, they can manually fix the email.

In fact - hear me out - what if we leave the the `deleted-at` section and stack on a new `restored-at` section as well! 😄

Reply:
Can't help but wonder how many toggles, deleted to/from restored it takes to break something. Just out of sheer curiosity than concern about someone actually trying it

Reply:
It looks like total email length can't exceed 255 characters.

Reply:
I changed it to tack on a `restored-at` instead now.